src/main/scala/akka/contrib/datareplication/ORMultiMap.scala

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.contrib.datareplication
+
+import akka.cluster.{UniqueAddress, Cluster}
+
+object ORMultiMap {
+
+  /**
+   * Provides an empty multimap.
+   */
+  val empty: ORMultiMap =
+    new ORMultiMap(ORMap.empty)
+
+  /**
+   * Construct a multimap.
+   */
+  def apply(map: ORMap): ORMultiMap =
+    new ORMultiMap(map)
+}
+
+/**
+ * An immutable multi-map implementation. This class wraps an
+ * [[ORMap]] with an [[ORSet]] for the map's value.
+ */
+@SerialVersionUID(1L)
+class ORMultiMap private (private[akka] val map: ORMap)
```

Comment:
I note that `ORMap` is a case class. I wasn't sure how this would benefit from being a case class, but of course, happy to change that.